lightfm/lightfm.py

```diff
@@ -315,6 +315,20 @@ def _get_lightfm_data(self):
 
         return lightfm_data
 
+    def _check_finite(self):
+
+        for parameter in (self.item_embeddings,
+                          self.item_biases,
+                          self.user_embeddings,
+                          self.user_biases):
+            # A sum of an array that contains non-finite values
+            # will also be non-finite, and we avoid creating a
+            # large boolean temporary.
+            if not np.isfinite(np.sum(parameter)):
+                raise ValueError("Not all estimated parameters are finite, "
+                                 "your model may have diverged. Try decreasing "
+                                 "the learning rate.")
```

Comment:
I think the exception should any contextual information it can, such as which epoch failed, what is the current learning-rate, and anything else that can help the caller resolve the divergence.